code/modules/power/apc/apc_main.dm

```diff
@@ -184,18 +176,24 @@
 			log_mapping("Duplicate APC created at [AREACOORD(src)] [area.type]. Original at [AREACOORD(area.apc)] [area.type].")
 		area.apc = src
 
-	update_appearance()
+	if(mapload)
+		has_electronics = APC_ELECTRONICS_SECURED
+		// is starting with a power cell installed, create it and set its charge level
+		if(cell_type)
+			cell = new cell_type(src)
+			cell.charge = start_charge * cell.maxcharge / 100 // (convert percentage to actual value)
+		make_terminal()
+		///This is how we test to ensure that mappers use the directional subtypes of APCs, rather than use the parent and pixel-shift it themselves.
+		if(abs(offset_old) != APC_PIXEL_OFFSET)
```

Comment:
Use TEST_ASSERT here

Reply:
While we are here it seems we are again initializing area and adding timer inside here 
https://github.com/tgstation/tgstation/blob/a29e9737038c2cec6216c3a4f2a41ef6ac0a3af8/code/modules/power/apc/apc_main.dm#L131-L139
Should I just move all the code inside New() to initialize()?